lib/stack_master/template_compiler.rb

```diff
@@ -2,12 +2,16 @@ module StackMaster
   class TemplateCompiler
     TemplateCompilationFailed = Class.new(RuntimeError)
 
-    def self.compile(config, template_dir, template_file_path, sparkle_pack_template, compile_time_parameters, compiler_options = {})
-      compiler = template_compiler_for_stack(template_file_path, sparkle_pack_template, config)
+    def self.compile(config, template_compiler, template_dir, template, compile_time_parameters, compiler_options = {})
+      compiler = if template_compiler
```

Comment:
Is this more readable? (I'm not sure)
```
compiler = @compilers.fetch(template_compiler) { |_| template_compiler_for_stack(template, config) }
```

Reply:
Hmmm, I think I'd rather not look in `@compilers` if `template_compiler` is nil 🤔